src/components/Modals/NewCourse/NewCourseModal.vue

```diff
@@ -109,48 +108,29 @@ export default Vue.extend({
       this.getReqsRelatedToCourse(result);
     },
     getReqsRelatedToCourse(selectedCourse: MatchingCourseSearchResult) {
-      const requirementsThatAllowDoubleCounting: string[] = [];
-      const relatedRequirements: RequirementWithID[] = [];
-      const potentialRequirements: RequirementWithID[] = [];
-      const groupedRequirements = store.state.groupedRequirementFulfillmentReport;
-
-      // parse through reqs object
-      for (let i = 0; i < groupedRequirements.length; i += 1) {
-        const subreqs = groupedRequirements[i].reqs.filter(
-          it => it.minCountFulfilled < it.minCountRequired
-        );
-        for (let j = 0; j < subreqs.length; j += 1) {
-          // requirements
+      const {
+        directlyRelatedRequirements,
+        selfCheckRequirements,
+      } = getRelatedUnfulfilledRequirements(
+        selectedCourse.id,
+        store.state.groupedRequirementFulfillmentReport,
+        store.state.toggleableRequirementChoices
+      );
 
-          const subRequirement = subreqs[j].requirement;
-          if (subRequirement.fulfilledBy === 'courses') {
-            const { courses } = subRequirement;
-            for (let k = 0; k < courses.length; k += 1) {
-              for (const [, ids] of Object.entries(courses[k])) {
-                if (ids.includes(selectedCourse.id)) {
-                  if (subRequirement.allowCourseDoubleCounting) {
-                    requirementsThatAllowDoubleCounting.push(subRequirement.name);
-                  } else {
-                    relatedRequirements.push({ id: subRequirement.id, name: subRequirement.name });
-                  }
-                  break;
-                }
-              }
-            }
-          }
-          // potential self-check requirements
-          if (subreqs[j].fulfilledBy === 'self-check') {
-            if (!subRequirement.allowCourseDoubleCounting) {
-              potentialRequirements.push({ id: subRequirement.id, name: subRequirement.name });
-            }
-          }
-          this.requirementsThatAllowDoubleCounting = requirementsThatAllowDoubleCounting;
-          this.relatedRequirements = relatedRequirements;
-          this.potentialRequirements = potentialRequirements;
-          this.selectedRequirementID =
-            relatedRequirements.length > 0 ? relatedRequirements[0].id : '';
+      const requirementsThatAllowDoubleCounting: string[] = [];
+      const relatedRequirements: RequirementWithIDSourceType[] = [];
+      directlyRelatedRequirements.forEach(it => {
+        if (it.allowCourseDoubleCounting) {
+          requirementsThatAllowDoubleCounting.push(it.name);
+        } else {
+          relatedRequirements.push(it);
         }
-      }
+      });
+
+      this.requirementsThatAllowDoubleCounting = requirementsThatAllowDoubleCounting;
+      this.relatedRequirements = relatedRequirements;
+      this.selfCheckRequirements = selfCheckRequirements;
```

Comment:
More general question about double-countable-reqs: Currently, do we apply the course we want to add to all the `requirementsThatAllowDoubleCounting`? Do we apply the course to any `relatedRequirements`?

Reply:
>  Currently, do we apply the course we want to add to all the `requirementsThatAllowDoubleCounting`?

Yes, we don't even need to store the choices. According to the graph algo, we first build a coarse graph in phase 1 where all possible edges are connected. The edge between a course and a requirement that allow double counting will never be removed in phase 3.

> That is, would the course be found under all those double-countable subrequirements in the Sidebar at the moment

Yes, since the edge exists in the graph, the `GroupedRequirementFulfillment` will also contain the course, so it will appear in side bar somewhere.

> Similarly, do we apply the course to any `relatedRequirements`

We will eventually, when the requirement choices are saved and integrated into the algo.

Reply:
Got it ty!